ui/src/app/services/error-handling.service.ts

```diff
@@ -0,0 +1,59 @@
+import {Injectable, NgZone} from '@angular/core';
+import {Observable} from 'rxjs/Observable';
+
+@Injectable()
+export class ErrorHandlingService {
+
+  // Expose "current user details" as an Observable
+  public fiveHundred: boolean;
+  public zero: boolean;
+
+  constructor(private zone: NgZone) {
+    this.fiveHundred = false;
+    this.zero = false;
+  }
+
+  public publishFiveHundred(): void {
+    this.fiveHundred = true;
+  }
+
+  public resolveFiveHundred(): void {
+    this.fiveHundred = false;
+  }
+
+  public publishZero(): void {
```

Comment:
I don't understand what "zero" means. SO says maybe this is a client-side convention for "no server response"? Not something I've run into before.

In general I'd prefer referencing the HTTP status codes by name -- publishServerError() and publishNoServerResponse(). Same idea in the template, I think we want a more verbose / non-technical description.

Reply:
Yup, I can do that.
Zero is the response it gets when it tries to ping the server and it finds nothing. For example, if I am pointing at my local environment, but the local dev API isn't running. (Could be good if we mistype a url for an API call or something.)